version: "3.7"
services:
  app:
    build:
      context: .
      dockerfile: Dockerfile.dev
    env_file:
      - .env
    environment:
      - FLASK_APP=app.factory:SubdomainDispatcher('localhost','default')
      - DATABASE_URL=postgresql://supportService:supportService@db/supportService
      - FLASK_ENV=development
      - FLASK_DEBUG=true
      - REDIS_HOST=cache
      - REDIS_URL=redis://cache:6379/0
      - LOG_TO_STDOUT=1
      - DATADOG_HOST=datadog-agent
      - DD_AGENT_HOST=datadog-agent
      - DD_AGENT_PORT=8126
    volumes:
      - .:/usr/src/app
    ports:
      - 5000:5000
    command: ["ddtrace-run", "python3", "app/factory.py", "--host=0.0.0.0"]
    depends_on:
      - cache

  cache:
    image: redis:alpine
    ports:
      - 6379:6379

  datadog-agent:
    image: datadog/agent:latest
    env_file:
      - .env
    environment:
      - DD_DOGSTATSD_NON_LOCAL_TRAFFIC=true
      - DD_APM_ENABLED=true
      - DD_APM_NON_LOCAL_TRAFFIC=true
    ports:
      - 8126:8126
    links:
      - app
